### DISCUSSION TITLE
Timeline on caching

### DISCUSSION THREAD

**Discussion by Sheikh-TDH** [okuramasafumi/alba#248] [Q&A] [answered]
Firstly I wanted to say migrating from AMS to Alba has been seamless and I really appreciate the work you have done.

But I wanted to ask if there was a timeline for when caching will be available?

Again appreciate the work you have done.

**Comment by okuramasafumi** [comment, marked as answer]
I once gave up implementing cache feature in Alba.
The primary reason is that we can implement cache feature by ourselves quite easily, like below:

```ruby
Rails.cache.fetch(cache_key_for_foos, expires_in: 5.minutes) do
  FooResource.new(foos).serialize
end
```

This way, within 5 minutes it returns from cache instead of serializing objects, that is much faster. Note that there's a possibility of incorrect result if you don't choose cache key carefully.

@Sheikh-TDH Did you compare performance between Alba and AMS with cache? I believe there's no much difference. I'd appreciate if you report the result. Thank you!
